backend/src/test/java/com/github/astridottmann/backend/controllers/CompareRoutesIntegrationTest.java

```diff
@@ -0,0 +1,199 @@
+package com.github.astridottmann.backend.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.astridottmann.backend.models.*;
+import com.github.astridottmann.backend.repositories.CompareRoutesRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class CompareRoutesIntegrationTest {
+    @Autowired
+    MockMvc mockMvc;
+    @Autowired
+    private CompareRoutesRepository compareRoutesRepository;
+    @Autowired
+    private ObjectMapper objectMapper;
+    private CompareRoutes testCompareRoutes;
+    private String testCompareRoutesJson;
+    private CompareRoutes testCompareRoutesWithoutId;
+    private String testCompareRoutesWithoutIdJson;
+    private String expectedBodyJson;
+    private String testComparedJson;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        testCompareRoutes = new CompareRoutes(
+                "999",
+                List.of(new Route("123",
+                                "Hamburg",
+                                "Frankfurt",
+                                492,
+                                1,
+                                false,
+                                new Car("car", 253.3, "petrol", "large"),
+                                268.93),
+                        new Route("456",
+                                "Hamburg",
+                                "Frankfurt",
+                                492,
+                                1,
+                                false,
+                                new PublicTransport("publicTransport", 46.0, "longDistance", "train"),
+                                45.26)),
```

Comment:
Use variables and add them to the list. Its easier to read:
```suggestion
               Car car = new Car("car", 253.3, "petrol", "large");
               Route routeA = new Route("123", "Hamburg", "Frankfurt", 492, 1, false, car, 268.93);
               
               PublicTransport publicTransport = new PublicTransport("publicTransport", 46.0, "longDistance", "train");
               Route routeB = new Route("456", "Hamburg", "Frankfurt", 492, 1, false, publicTransport, 45.26)
               
                List.of(routeA, routeB),
```

Reply:
very good idea! changed it!